lib/index.d.ts

```diff
@@ -736,8 +736,8 @@ declare namespace Joi {
         [key in keyof TSchema]?: SchemaLike | SchemaLike[];
     } 
 
-    type StrictSchemaMap<TSchema = any> =  {
-        [key in keyof TSchema]-?: ObjectPropertiesSchema<TSchema[key]>
+    type StrictSchemaMap<TSchema = any> = {
+        [key in keyof TSchema]-?: ObjectPropertiesSchema<TSchema[key]> | AlternativesSchema
```

Comment:
To me it's the wrong place to do it. I've tried it by modifying `ObjectPropertiesSchema` alone and it seems to work, let me know what you think.
```ts
    type IsUnion<T, U extends T = T> =
      T extends unknown ? [U] extends [T] ? false : true : false;

    type ObjectPropertiesSchema<T = any> =
        true extends IsUnion<Exclude<T, undefined>>
        ? Joi.AlternativesSchema
        : T extends NullableType<string>
        ? Joi.StringSchema
        : T extends NullableType<number>
        ? Joi.NumberSchema
        : T extends NullableType<bigint>
        ? Joi.NumberSchema
        : T extends NullableType<boolean>
        ? Joi.BooleanSchema
        : T extends NullableType<Date>
        ? Joi.DateSchema
        : T extends NullableType<Array<any>>
        ? Joi.ArraySchema
        : T extends NullableType<object>
        ? (StrictSchemaMap<T> | ObjectSchema<T>)
        : never
```